tests/unit/test-cmd/test.lint.js

```diff
@@ -3,157 +3,39 @@ import {it, describe} from 'mocha';
 
 describe('lint', () => {
 
-  function setUp({createLinter, createFileFilter}: setUpParams = {}) {
-    const lintResult = '<lint.run() result placeholder>';
-    const runLinter = sinon.spy(() => Promise.resolve(lintResult));
-    if (!createLinter) {
-      createLinter = sinon.spy(() => {
-        return {run: runLinter};
+  it('creates and runs a linter', async () => {
+    const fakeLinter = sinon.spy(() => lintResult);
+    const lintResult = {
+      run: sinon.spy(() => Promise.resolve()),
+    };
+    const actualLintResult = await lintCommand({
+      sourceDir: '/fake/source/dir',
+    }, {
+      linter: fakeLinter,
+    });
+    assert.equal(actualLintResult, lintResult);
+    assert.equal(fakeLinter.called, true);
+  });
+
+  it('fails when the linter fails', async () => {
+    const fakeLinter =
+      sinon.spy(() => Promise.reject(new Error('linter error')));
+    let exception;
+    try {
+      await lintCommand({
+        sourceDir: '/fake/source/dir',
+      }, {
+        linter: fakeLinter,
       });
+    } catch (linterError) {
+      exception = linterError;
     }
-    return {
-      lintResult,
-      createLinter,
-      runLinter,
-      lint: (params = {}, options = {}) => {
-        return defaultLintCommand({
-          sourceDir: '/fake/source/dir',
-          ...params,
-        }, {
-          createLinter,
-          createFileFilter,
-          ...options,
-        });
-      },
-    };
-  }
-
-  it('creates and runs a linter', () => {
-    const {lint, createLinter, runLinter, lintResult} = setUp();
-    return lint().then((actualLintResult) => {
-      assert.equal(actualLintResult, lintResult);
-      assert.equal(createLinter.called, true);
-      assert.equal(runLinter.called, true);
-    });
-  });
-
-  it('fails when the linter fails', () => {
-    const createLinter = () => {
-      return {
-        run: () => Promise.reject(new Error('some error from the linter')),
-      };
-    };
-    const {lint} = setUp({createLinter});
-    return lint().then(makeSureItFails(), (error) => {
-      assert.match(error.message, /error from the linter/);
-    });
-  });
-
-  it('runs as a binary', () => {
-    const {lint, createLinter} = setUp();
-    return lint().then(() => {
-      const args = createLinter.firstCall.args[0];
-      assert.equal(args.runAsBinary, true);
-    });
-  });
-
-  it('sets runAsBinary according shouldExitProgram option', () => {
-    const {lint, createLinter} = setUp();
-    return lint({}, {shouldExitProgram: false}).then(() => {
-      const args = createLinter.firstCall.args[0];
-      assert.strictEqual(args.runAsBinary, false);
-    });
-  });
-
-  it('passes sourceDir to the linter', () => {
-    const {lint, createLinter} = setUp();
-    return lint({sourceDir: '/some/path'}).then(() => {
-      const config = createLinter.firstCall.args[0].config;
-      assert.equal(config._[0], '/some/path');
-    });
-  });
 
-  it('passes warningsAsErrors to the linter', () => {
```

Comment:
I cannot find this test case anymore, it should probably be one of the tests of the new linter module.